game-core/src/main/java/tools/map/making/MapPropertyWrapper.java

```diff
@@ -40,38 +40,36 @@
  * @param <T> parameters can be: Boolean, String, Integer, Double, Color, File, Collection, Map
  */
 @Log
-public class MapPropertyWrapper<T> extends AbstractEditableProperty {
+public class MapPropertyWrapper<T> extends AbstractEditableProperty<T> {
   private static final long serialVersionUID = 6406798101396215624L;
-  private final IEditableProperty property;
+  private final IEditableProperty<T> property;
   private final Method setter;
 
-  // TODO: remove the `getter` field when we can, kept around for serialization compatibility
-  @SuppressWarnings("unused")
-  private final Method getter = null;
 
+  @SuppressWarnings("unchecked")
   private MapPropertyWrapper(final String name, final String description, final T defaultValue, final Method setter) {
     super(name, description);
     this.setter = setter;
 
 
     if (defaultValue instanceof Boolean) {
-      property = new BooleanProperty(name, description, ((Boolean) defaultValue));
+      property = (IEditableProperty<T>) new BooleanProperty(name, description, ((Boolean) defaultValue));
```

Comment:
Just an idea to possibly get rid of this repeated cast...  Could we first assign the expression to a local variable of type `IEditableProperty<?>`, and then do the `IEditableProperty<T>` cast to the field at the end of the constructor?